Add `as_tensor_type` method to `TensorTable` for framework column conversion

oliverholworthy opened the pull request:
Adds a convenience method `to` to the `TensorTable` for converting columns between framework types.

## Example usage

```python
table = TensorTable({"feature": np.array([1, 2, 3])})

table.as_tensor_type(torch.Tensor).to_dict()
# => {'a': torch.tensor([1, 2, 3])}

table.as_tensor_type(TorchColumn).to_dict()
# => {'a': torch.tensor([1, 2, 3])}
```

```python
table = TensorTable({"feature": np.array([1, 2, 3])})
table.as_tensor_type("unknown")
# => Raises `ValueError`
# ValueError: tensor_type argument must be a type. Received: <class 'str'> 
# Supported values are: 'numpy.ndarray', 'torch.Tensor', 'tensorflow.python.framework.ops.Tensor'. 
# Or TensorColumn Types: 'merlin.table.NumpyColumn', 'merlin.table.TorchColumn', 'merlin.table.TensorflowColumn'
```

karlhigley commented:
Those all sound like good suggestions to me, and I can see any of them working okay

oliverholworthy commented:
I've re-requested your review @karlhigley since I've made some updates that change things quite a bit since you last reviewed. 

- renamed `.to` to `.as_tensor_type` 
- changed the argument type from a string to `Union[Type, TensorColumn]`
